Dapper/SqlMapper.DapperRow.cs

```diff
@@ -208,6 +209,41 @@ private object SetValue(string key, object value, bool isAdd)
             }
 
             #endregion
+
+
+            #region Implementation of IReadOnlyDictionary<string,object>
+
+
+            int IReadOnlyCollection<KeyValuePair<string, object>>.Count
+            {
+                get
+                {
+                    return values.Count(t => !(t is DeadValue));
+                }
+            }
+
+            bool IReadOnlyDictionary<string, object>.ContainsKey(string key)
+            {
+                int index = table.IndexOfName(key);
+                return index >= 0 && index < values.Length && !(values[index] is DeadValue);
+            }
+
+            object IReadOnlyDictionary<string, object>.this[string key]
+            {
+                get { TryGetValue(key, out object val); return val; }
+            }
+
+            IEnumerable<string> IReadOnlyDictionary<string, object>.Keys
+            {
+                get { return this.Select(kv => kv.Key); }
+            }
+
+            IEnumerable<object> IReadOnlyDictionary<string, object>.Values
+            {
+                get { return this.Select(kv => kv.Value); }

```

Comment:
possibly `return values;`, although similar concerns to `Keys` - probably best to leave this "as is", too!